app/src/main/java/com/parasoft/demoapp/component/OrderDialog.java

```diff
@@ -190,27 +194,31 @@ public void onFailure(@NonNull Call<ResultResponse<OrderResponse>> call, @NonNul
             });
     }
 
-    private void updateOrderStatus(OrderResponse oldOrderInfo) {
-        OrderStatusRequest orderStatusRequest = new OrderStatusRequest();
-        orderStatusRequest.setStatus(oldOrderInfo.getStatus());
-        orderStatusRequest.setReviewedByAPV(true);
-
-        pdaService.getClient(ApiInterface.class).orderDetails(orderNumber, orderStatusRequest)
+    private void updateOrderDetails(OrderStatusRequest orderStatusRequest) {
+        pdaService.getClient(ApiInterface.class).updateOrderDetails(orderNumber, orderStatusRequest)
                 .enqueue(new Callback<ResultResponse<OrderResponse>>() {
                     @Override
                     public void onResponse(@NonNull Call<ResultResponse<OrderResponse>> call, @NonNull Response<ResultResponse<OrderResponse>> response) {
-                        if (response.code() != 200) {
+                        if(response.code() == 200) {
+                            orderInfo = response.body().getData();
+                            if (saveButton.isEnabled()) {
```

Comment:
I think we should disable "Save" and "Cancel" button while waiting for the response and only enable "Save" and "Cancel" when the call is failed. So we can fix the server connection issue and use "Save" to submit again or "Cancel" the submit.
Please do not use "if (saveButton.isEnabled())" here. Instead, pass in a boolean flag at line 197 to control this.
Also please verify whether following case is doing proper things:
1. Open an open order's detail dialog
2. Take down the PDA web app
3. Try to submit a response, and right after that, click "x" at top.
4. See if this would make the app force quit or behave incorrectly.